samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/print/PDFAssessmentBean.java

```diff
@@ -786,6 +753,92 @@ else if(TypeIfc.CALCULATED_QUESTION.equals( item.getItemData().getTypeId() )){
 		return contentBuffer.toString();
 	}
 
+	public String getContentQuestionImageMap(ItemContentsBean item, PrintSettingsBean printSetting, boolean renderMap) {
+		StringBuffer contentBufferMap = new StringBuffer("");
+		//look for path on metadata
+		String imsrc=item.getItemData().getItemMetaDataByLabel("IMAGE_MAP_SRC");
+		if (imsrc==null)
+			imsrc="";
+
+		imsrc=imsrc.replaceAll(" ", "%20");
+
+		String ext = "";
+		if (imsrc.lastIndexOf('.') > 0)
+			ext = imsrc.substring(imsrc.lastIndexOf('.')+1);
+
+		BufferedImage img_in=null;
+		int w=-1;
+		int h=-1;
+		try {
+			URL url = new URL(ServerConfigurationService.getServerUrl()+imsrc);
+			img_in = ImageIO.read(url);
+			w = img_in.getWidth(null);
+			h = img_in.getHeight(null);
+		} catch (IOException e) {
+			log.error(e.getMessage(), e);
+		}
+
+		if (renderMap) {
+
+			Color c = new Color(27, 148, 224, 80);
+			java.awt.Font font = new java.awt.Font("Serif", java.awt.Font.BOLD, 10);
+
+			//print areas over the image
+			BufferedImage img_out = new BufferedImage(w, h, BufferedImage.TYPE_INT_RGB);
+			try {
+				Graphics g = img_out.getGraphics();
+				g.setFont(font);
+				g.drawImage(img_in, 0, 0, null);
+
+				List<ItemTextIfc> question = item.getItemData().getItemTextArraySorted();
+				for (int k=0; k<question.size(); k++) {
+					PublishedItemText itemtext = (PublishedItemText)question.get(k);
+					List<PublishedAnswer> answers=itemtext.getAnswerArray();
+					PublishedAnswer answer = (PublishedAnswer)answers.get(0);
+
+					String area = answer.getText();
+					Integer areax1=Integer.valueOf(area.substring(area.indexOf("\"x1\":")+5,area.indexOf(",", area.indexOf("\"x1\":"))));
+					Integer areay1=Integer.valueOf(area.substring(area.indexOf("\"y1\":")+5,area.indexOf(",", area.indexOf("\"y1\":"))));
+					Integer areax2=Integer.valueOf(area.substring(area.indexOf("\"x2\":")+5,area.indexOf(",", area.indexOf("\"x2\":"))));
+					Integer areay2=Integer.valueOf(area.substring(area.indexOf("\"y2\":")+5,area.indexOf("}", area.indexOf("\"y2\":"))));
+
+					g.setColor(c);
+					g.fillRect(areax1, areay1, (areax2-areax1), (areay2-areay1));
+
+					g.setColor(Color.WHITE);
+					g.drawRect(areax1, areay1, (areax2-areax1), (areay2-areay1));
+
+					g.setColor(Color.BLACK);
+					g.drawString(""+(k+1), areax2-13, areay1+13);
+				}
+				g.dispose();
+				img_in=img_out;
+			} catch (Exception e) {
+				log.error(e.getMessage(), e);
+			}
+		}
+
+		String destSrc = "temp://";
```

Comment:
The temp file should be cleaned up:
```
try {
	File temp = File.createTempFile("imgtemp", "." + ext);
	temp.deleteOnExit();
	ImageIO.write(img_out, ext, temp);
	contentBuffer.append("  <img src=\"temp://").append(temp.getCanonicalPath()).append("\"/> ");
} catch (IOException e) {
	log.warn("Could not create a temporary file for pdf image {}", e.getMessage(), e);
}
```

Reply:
This file should actually be cleanup once the PDF is written but the above is better than no cleanup.

Reply:
@ern Done!

Reply:
@ern Is everything ok?